go-controller/pkg/node/gateway_localnet_linux_test.go

```diff
@@ -535,7 +531,6 @@ var _ = Describe("Node Operations", func() {
 
 		It("inits iptables rules and openflows with LoadBalancer where ETP=local, LGW mode", func() {
 			app.Action = func(ctx *cli.Context) error {
-				externalIP := "1.1.1.1"
```

Comment:
@fedepaol : why are we removing the testing portion for externalIPs? A service can have both `spec.ExternalIPs` and `spec.status.ingress.IP` set right or are they mutually exclusive for a LB type ? - even then better to unit test it because I recently had a bug where a metalLB user tried to set the same IP for ingress.IP and externalIP :). 

We should add this back because if those two IPs are different the flows would also be different...seems like we removed the testing for that unless I am missing something, so in the end we should have 2 iptable rules in externalIPs chain.

Reply:
Yeah, I was torn about this.
I don't think it's common to have a loadbalancer service where the user asks also for ExternalIP. The most common scenario is to create the service and let the loadbalancer fill `service.Status.LoadBalancer.Ingress`. In load balancer services, if the user wants to ask for an explicit ip on the service, he must fills `Spec.LoadBalancerIP`.
So, my understanding is that externalIPs + loadbalancer is a corner case (not sure if it's even possible) and I thought it is more clear to have the more general case tested. 
I also considered not using externalIPs at all in producing the iptables rules for a lb service, but this felt a bit safer.

Said that, I will re-add the externalips to the test cases where I removed them.

Reply:
@tssurya restored it